packages/magic-entries-webpack-plugin/src/test/utilities.ts

```diff
@@ -0,0 +1,48 @@
+import * as path from 'path';
+
+import * as fs from 'fs-extra';
+
+export class Workspace {
+  constructor(public readonly root: string) {}
+
+  resolvePath(...parts: string[]) {
+    return path.resolve(this.root, ...parts);
+  }
+
+  buildPath(...parts: string[]) {
+    return this.resolvePath('build', ...parts);
+  }
+
+  async write(file: string, contents: string) {
+    const filePath = this.resolvePath(file);
+    await fs.mkdirp(path.dirname(filePath));
+    await fs.writeFile(filePath, contents);
+  }
+
+  async cleanup() {
+    await fs.emptyDir(this.root);
+    await fs.remove(this.root);
+  }
+}
+
+const rootFixtureDirectory = path.resolve(__dirname, '../fixtures');
+
+export async function createWorkspace({name}: {name: string}) {
+  const fixtureDirectory = path.join(rootFixtureDirectory, name);
+  await fs.mkdirp(fixtureDirectory);
+  await fs.writeFile(path.join(fixtureDirectory, '.gitignore'), '*');
```

Comment:
It seems a bit strange to `.gitignore` all these manually. Maybe use a naming convention for the fixture directories and `.gitignore` them in quilts root `.gitignore`. If nested `.gitignore`s is a thing maybe a `.gitignore` in this plugin directory.

Reply:
We have the same util as this in a couple other libraries, so I think manually gitignoring all of them might be weird. I don't really see a problem with us writing the gitignore, as we're already messing around in the filesystem in these folders anyway